Dockerfile

```diff
@@ -0,0 +1,37 @@
+# syntax = docker/dockerfile:1.4
```

Comment:
Do we want this to be the "root" Dockerfile for the project? Would it make sense to have a `/docker` or `/images` or `/containers` directory where we can house multiple Dockerfiles for different reasons? What about plumbing the docker commands through to the Makefile and having an `$OCI` variable so we can build with out container tools as well?

Something like this:

```make 
OCI=docker
$(OCI) build -t $(NAME) .
```

Reply:
You beat me to it - I was mid writing up a comment and was going to ask if this was the desired location. I'll move it and add the docker commands via the make file too.

Reply:
I'm not using any native docker commands in the pipeline, just pre-defined GHA jobs. I have added the ability to build the docker container to the Makefile.